### ISSUE TITLE
Failed to reproduce the few-shot results.

### ISSUE THREAD

**Issue by Caox1ngge** [xmed-lab/NumCLIP#5]
When I tried running the commands below in the terminal, I got a 2-shot MAE of 4.00. Why is this happening?
python scripts/run.py \
--config configs/default_test.yaml \
--config configs/base_cfgs/data_cfg/datasets/morph/morph.yaml \
--config configs/base_cfgs/runner_cfg/model/image_encoder/clip-vitb16.yaml \
--config configs/base_cfgs/runner_cfg/model/text_encoder/clip-vitb16-cntprt.yaml \
--config configs/base_cfgs/runner_cfg/model/prompt_learner/init_context/init-context-morph_class.yaml \
--config configs/base_cfgs/runner_cfg/model/prompt_learner/num_ranks_5.yaml \
--config configs/base_cfgs/runner_cfg/model/prompt_learner/plain-prompt-learner.yaml \
--config configs/base_cfgs/runner_cfg/optim_sched/prompt_learner/tune-rank.yaml \
--config configs/base_cfgs/runner_cfg/optim_sched/image_encoder/tune-image.yaml \
--config configs/base_cfgs/runner_cfg/model/regclipssr.yaml \
--config configs/base_cfgs/runner_cfg/model/prompt_learner/init_rank/init-rank-morph_class.yaml
--config configs/base_cfgs/runner_cfg/init_weight.yaml
--config configs/base_cfgs/data_cfg/few_shots/num-shots-2.yaml
--config configs/base_cfgs/data_cfg/label_distribution_shift/num_topk_scaled_class/num-topk-scaled-class-40.yaml \
--config configs/base_cfgs/data_cfg/label_distribution_shift/scale_factor/scale-factor-01.yaml

**Comment by duyao-art** [comment]
Thank you for your interest in our work. Could you please provide more details about your implementation?

From a quick check, if you are working under a few-shot setting, it may not be necessary to set the label_distribution_shift parameter. You may want to revisit this configuration.

**Comment by Caox1ngge** [comment]
Thank you very much for your response. I did not modify the original code at all. I ran the experiments in the terminal exactly following the commands in the run_regclipssr.sh script. The 1-shot result I got was 3.66, which is very close to the value reported in the paper. However, the results for 2-shot and 4-shot were both around 4.x. I only changed the part marked in red in the config line for num-shots-*.yaml.

**Comment by Caox1ngge** [comment]
<img width="1293" height="476" alt="Image" src="https://github.com/user-attachments/assets/6b65b686-3e6a-4fb4-8b8b-06f0f95b5b16" />

**Comment by Caox1ngge** [comment]
runner_cfg:
  seed: 114514
  output_dir: results/norecord/ViTb16-SSR/Morph/
  optimizer_and_scheduler_cfg:
    param_dict_cfg:
      lr_prompt_learner_context: 0
      lr_prompt_learner_ranks: 0.0001
      lr_image_encoder: 0.0001
      lr_image_adapter: 0.00001
      lr_regressor: 0.00001
      lr_text_encoder: 0.0
      lr_logit_scale: 0.0
      # lr_adapter: 0.0
      staged_lr_image_encoder:
        new_layers: []
        base_lr_mult: 1.0
        new_lr_mult: 1.0
    optimizer_cfg:
      optimizer_name: radam
      lr: 0.0001
      weight_decay: 0.0
      momentum: 0.9
      sgd_dampening: 0.0
      sgd_nesterov: False
      rmsprop_alpha: 0.99
      adam_beta1: 0.9
      adam_beta2: 0.999
      staged_lr: null
      lookahead: False
    lr_scheduler_cfg:
      lr_scheduler_name: multi_step
      stepsize:
        - 60
      gamma: 0.1
      max_epochs: &max_epochs 200
      warmup_epoch: 0
      warmup_cons_lr: 1.0e-05
      warmup_min_lr: 1.0e-05
      warmup_type: constant
      warmup_recount: True
  load_weights_cfg:
    init_model_weights: null
    init_prompt_learner_weights: null
    init_image_encoder_weights: null
    init_image_adapter_weights: null
    init_text_encoder_weights: null
    # init_adapter_weights: null
    init_regressor_weights: null
  loss_weights:
    ce_loss: 1.0
    kl_loss: 1.0
    reg_loss: 1.0
  model_cfg:
    type: RegCLIPSSR
    prompt_learner_cfg:
      type: PlainPromptLearner
      num_ranks: 5
      num_base_ranks: 3
      num_tokens_per_rank: 1
      num_context_tokens: 10
      rank_tokens_position: tail
      init_rank_path: data/classnames_class.txt
      init_context: 'age estimation: the person is a'
      rank_specific_context: false
      interpolation_type: linear
    text_encoder_name: ViT-B/16
    image_encoder_name: ViT-B/16
data_cfg:
  train_images_root: data/MORPH
  val_images_root: data/MORPH
  test_images_root: data/MORPH
  train_data_file: data/MORPH/data_list/ET_proto_train.txt
  val_data_file: data/MORPH/data_list/ET_proto_val.txt
  test_data_file: data/MORPH/data_list/ET_proto_val.txt
  few_shot:
    num_shots: 2
  label_distributed_shift:
    num_topk_scaled_class: 40
    scale_factor: 0.1
  use_long_tail: false
  transforms_cfg:
    input_transforms:
      - random_resized_crop
      - random_hflip
      - normalize
    input_resize:
      - 256
      - 256
    input_size:
      - 224
      - 224
    pixel_mean:
      - 0.485
      - 0.456
      - 0.406
    pixel_std:
      - 0.229
      - 0.224
      - 0.225
  train_dataloder_cfg:
    num_workers: 8
    batch_size: 32
    shuffle: true
  eval_dataloder_cfg:
    num_workers: 8
    batch_size: 32
    shuffle: false
trainer_cfg:
  max_epochs: *max_epochs
  precision: 16
  accelerator: gpu
  devices: 1
  check_val_every_n_epoch: 1
  num_sanity_val_steps: 1
  fast_dev_run: false
test_only: False

**Comment by duyao-art** [comment]
few_shot:
num_shots: 2
label_distributed_shift:
num_topk_scaled_class: 40
scale_factor: 0.1    

Based on your configuration, it seems that both few-shot training and label distribution shift are enabled simultaneously.

If your goal is to evaluate the few-shot setting only, you may consider disabling the label distribution shift module. In particular, you can comment out the following line:
# --config configs/base_cfgs/data_cfg/label_distribution_shift/scale_factor/scale-factor-01.yaml
and then rerun the experiment.

**Comment by Caox1ngge** [comment]
<img width="1623" height="234" alt="Image" src="https://github.com/user-attachments/assets/4c0481b4-1469-4432-a18e-2a26fb1a634e" />

**Comment by Caox1ngge** [comment]
The result has improved to 3.39, but it is still far from the original paper's 3.17. Are there any other details I need to modify or pay attention to?

**Comment by Caox1ngge** [comment]
In addition, under the 4-shot experimental setting, we only achieved an MAE of 3.60. Could you please advise on the possible underlying reasons? Alternatively, would it be possible for you to share your 4-shot configuration for our reference?
